src/simulations/initial_conditions.jl

```diff
@@ -325,3 +325,68 @@ function make_set_initial_state_from_file(
     end
     return set_ic!
 end
+
+function set_soil_initial_conditions_from_temperature_and_total_water!(
+    Y,
+    subsurface_space,
+    soil_ic_path,
+    soil;
+    water_varname = "swvl",
+    temp_varname = "stl",
+    regridder_type = regridder_type,
+    extrapolation_bc = extrapolation_bc,
+    interpolation_method = interpolation_method,
+)
+    total_water_content = SpaceVaryingInput(
+        soil_ic_path,
+        water_varname,
+        subsurface_space;
+        regridder_type,
+        regridder_kwargs = (; extrapolation_bc, interpolation_method),
+    )
+    raw_data = NCDataset(soil_ic_path)
+    raw_temp_data = raw_data[temp_varname][:]
+    FT = eltype(Y.soil.ϑ_l)
+    T_bounds = FT.(extrema(raw_temp_data[raw_temp_data .> 0]))
```

Comment:
this may be specific to ERA5, which has T_soil = 0 over the ocean. Because our interpolation is not mask aware, and because the ClimaLand and ERA5 masks may not align exactly, we may end up with nonphysical values in `total_water_content` over locations that ClimaLand considers to be land. This clips temperature to the bounds over land in the data.

Reply:
(but this could lead to some locations that ERA5 calls ocean, but ClimaLand calls land, being set to the minimum value of T in the data, when in reality that location is not in a cold area)

Reply:
Maybe add a comment on this may result in too cold soil temperature in some areas?